cypress/integration/unit_tests/authenticate_test.js

```diff
@@ -0,0 +1,214 @@
+  const getStub = cy.stub(ee.data, 'getIamPolicy').callsFake(() => {
+    getResolveFunction();
+    return Promise.resolve(publicPolicy);
+  });
+  const setIamStub = cy.stub(ee.data, 'setIamPolicy');
+  waitForTrackAndAssertNormalStubs();
+  cy.wrap(getPromise).then(() => {
+    expect(getStub).to.be.calledOnce;
+    expect(getStub).to.be.calledWith(primaryPath);
+    expect(setIamStub).to.not.be.called;
+  });
+});
+
+/**
+ * Reaches deep into EarthEngine internals and constructs IamPolicy objects for
+ * returning in {@link ee.data.getIamPolicy} stubs. Calling that method doesn't
+ * seem to work in tests, even for assets we normally have write access to.
+ *
+ * This is so gross, but it allows us to construct genuine Policy objects so
+ * that we know our code can deal with them. If names change, test will break,
+ * but should be able to find new ones.
+ * TODO(janakr): See what EE says about how to better construct policies.
+ *
+ * @return {Object} Collection of policies for use in tests
+ */
+function getPolicies() {
+  const PolicyConstructor = module$exports$eeapiclient$ee_api_client.Policy;
+  const BindingConstructor = module$exports$eeapiclient$ee_api_client.Binding;
+  const ownerBinding = new BindingConstructor({
+    role: 'roles/owner',
+    members: ['user:gd-earthengine-user@givedirectly.org'],
+  });
+  const privateReaderBinding = new BindingConstructor({
+    role: 'roles/viewer',
+    members: ['user:gd-earthengine-user@givedirectly.org'],
+  });
+  const allReadWithReaderBinding = new BindingConstructor({
+    role: 'roles/viewer',
+    members: ['user:gd-earthengine-user@givedirectly.org', 'allUsers'],
+  });
+  const allReadBinding =
+      new BindingConstructor({role: 'roles/viewer', members: ['allUsers']});
+  return {
+    noReadPolicy: new PolicyConstructor({bindings: [ownerBinding]}),
+    privateReadPolicy:
+        new PolicyConstructor({bindings: [ownerBinding, privateReaderBinding]}),
+    publicPolicy:
+        new PolicyConstructor({bindings: [ownerBinding, allReadBinding]}),
+    publicPolicyWithReader: new PolicyConstructor(
+        {bindings: [ownerBinding, allReadWithReaderBinding]},
+        ),
+  };
+}
+
+it('Does not try to make score assets readable when not gd user', () => {
+  cy.get('@gapiEmail')
+      .then((emailStub) => emailStub.returns('some-user@givedirectly.org'));
+  const getPolicyStub = cy.stub(ee.data, 'getIamPolicy');
+  waitForTrackAndAssertNormalStubs();
+  // Give other thread a chance to call listAssets if that's going to happen.
+  cy.wait(0).then(() => expect(getPolicyStub).to.not.be.called);
+});
+
+/**
+ * Asserts that all the library calls we expect to encounter have happened. For
+ * conciseness, we use the `this` mode of accessing Cypress-aliased variables
+ * (https://docs.cypress.io/guides/core-concepts/variables-and-aliases.html#Sharing-Context).
+ * This means that this function cannot be called from within an arrow function.
+ * @this Context
```

Comment:
Any prefixes you can add to this Context to describe more where it's coming from

Reply:
Done, sorry, got lazy.